### PR TITLE
MAISTRA-578 cleanup ClusterRole for istio-operator

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

**Comment by rcernich** [comment]
@dmitri-d, this is a cleaned up version of the `ClusterRole` required by the operator.  I still need to run some tests through bookinfo to verify that everything is working correctly.  Note, there are still a few items that need to be removed, but those require changes to some of the components.

File: deploy/servicemesh-operator.yaml [hunk 1]
```diff
@@ -127,23 +127,27 @@ rules:
...
+  - namespaces
   - persistentvolumeclaims
-  - events
-  - configmaps
+  - pods
+  - replicationcontrollers
   - secrets
   - serviceaccounts
-  - namespaces
-  - routes
```
Comment (brian-avery) [inline deploy/servicemesh-operator.yaml:139]: is routes still needed?

File: deploy/maistra-operator.yaml [hunk 2]
```diff
@@ -127,23 +127,27 @@ rules:
...
+  - namespaces
   - persistentvolumeclaims
-  - events
-  - configmaps
+  - pods
+  - replicationcontrollers
   - secrets
   - serviceaccounts
-  - namespaces
-  - routes
```
Comment (brian-avery) [inline deploy/maistra-operator.yaml:139]: is routes still needed?

File: deploy/servicemesh-operator.yaml (hunk 1 above)
Comment (rcernich) [reply deploy/servicemesh-operator.yaml:139]: It's namespaced now, route.openshift.io/v1 instead of just v1.

**Comment by rcernich** [comment]
Also removes jaeger-agent service account, unsupported charts, and most instances of use anyuid scc.

**Review by jwendell — APPROVED** [review]

**Review by brian-avery — APPROVED** [review]

**Pull request merged**